### PR TITLE
fix(NVI): stuck PointerOver state

### LINKED ISSUE #7150: [Gallery][iOS] Multiple controls display as selected in the Navigation view.
### Current behavior

https://user-images.githubusercontent.com/74563602/134068293-95655914-7d40-460f-acb9-e4db2e2f0b7a.MOV

### Expected behavior

Only one selected control should display in the navigation view. 

### How to reproduce it (as minimally and precisely as possible)

1. Open Uno Gallery APP
2. Click Burger Menu on the left top right .
3. Click on multiple controls on the "component" list..
4. Observe that multiple controls display as selected in the Navigation view.

### Workaround

_No response_

### Works on UWP/WinUI

_No response_

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

iOS

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable): closes #7150

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Tapping on a NavigationViewItem with some dragging can result in the NVI to be stuck in the PointerOver visual-state, until next pointer interaction.

## What is the new behavior?

Make sure NVI's pointer-over flag is always cleared properly.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
This bug seems to only manifest on a custom style (MaterialMUXNavigationViewItemPresenterStyleWhenOnLeftPane) in Uno.Themes.

### CODE REVIEW

**Review by jeromelaban — COMMENTED** [review]
Can you add a UI test for this? Also, do you know which PR of Uno introduced this change?

**Comment by jeromelaban** [comment]
@dr1rrb

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewItem.cs [hunk 1]
```diff
@@ -1108,7 +1108,12 @@ private void ProcessPointerCanceled(PointerRoutedEventArgs args)
 			// What this flag tracks is complicated because of the NavigationView sub items and the m_capturedPointers that are being tracked..
 			// We do this check because PointerCaptureLost can sometimes take the place of PointerReleased events.
 			// In these cases we need to test if the pointer is over the item to maintain the proper state.
-			if (IsOutOfControlBounds(args.GetCurrentPoint(this).Position))
+			if (IsOutOfControlBounds(args.GetCurrentPoint(this).Position)
+#if HAS_UNO
+				// UNO#7150 workaround: pointer can be still within when the events are raised
+				|| (args.Pointer.PointerDeviceType == PointerDeviceType.Touch && !args.Pointer.IsInContact)
+#endif
```
Comment (dr1rrb) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationViewItem.cs:1115]: This change makes sens, I do had to do something similar for pointer over. I'm wondering however why this change is needed only for Uno, the issue should be the same on UWP. Are we able to repro with touch on XCG?

Note: As `Touch` is the common pointer on platform that do have the less power, you should inverse the 2 conditions: the `args.GetCurrentPoint(this)` can be pretty expensive, if we can avoid it, we should!

File: src/SamplesApp/UITests.Shared/Helpers/VisualTreeHelperEx.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+using System.Text;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Media;
+
+namespace UITests.Helpers
+{
+	/// <summary>
+	/// Contains various methods to debug the visual tree
+	/// </summary>
+	internal static class VisualTreeHelperEx
```
Comment (dr1rrb) [inline src/SamplesApp/UITests.Shared/Helpers/VisualTreeHelperEx.cs:13]: Why do we need one more version of "GetParent" / "GetChild" / "GetDebugIdentifier" / "ShowLocalVisualTree"? We already have way to much versions / variants of this kind of helpers. We should not add another set while we are still trying to remove the deprecated ones ... unless those helpers are coming from MUX?

**Pull request closed without merging**